OptionSelector/MainTableViewController.m

```diff
@@ -0,0 +1,139 @@
+                                   @"Eel",
+                                   @"Caecilian"];
+    
+    
+    
+    CQCategory *fishCategory = [[CQCategory alloc] init];
+    
+    fishCategory.name = @"Fish";
+    fishCategory.options = @[@"Catfish",
+                             @"Sea Trout",
+                             @"Sunfish",
+                             @"Salmon",
+                             @"Perch",
+                             @"Halinbut"];
+    
+    
+    CQCategory *reptilesCategory = [[CQCategory alloc] init];
+    
+    reptilesCategory.name = @"Reptiles";
+    reptilesCategory.options = @[@"Snake",
+                                   @"Iguana",
+                                   @"Alligator",
+                                   @"Bearded Dragon",
+                                   @"Savannah Monitor",
+                                   @"Tortoise"];
+    
+    
+    
+    CQCategory *mammalsCategory = [[CQCategory alloc] init];
+    
+    mammalsCategory.name = @"Mammals";
+    mammalsCategory.options = @[@"Cat",
+                                @"Dog",
+                                @"Ferret",
+                                @"Skunk",
+                                @"Rat",
+                                @"Sugar Glider"];
+    
+    
+    
+    
+    CQCategory *birdsCategory = [[CQCategory alloc] init];
+    
+    birdsCategory.name = @"Birds";
+    birdsCategory.options = @[@"African Grey",
+                              @"Cockatiel",
+                              @"Dove",
+                              @"Falcon",
+                              @"Eagle",
+                              @"Sparrow"];
+    
+    self.animalCategories = @[invertebratesCategory, amphibiansCategory, fishCategory, reptilesCategory, mammalsCategory, birdsCategory];
+    
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+    [self.tableView reloadData];
+}
+
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.animalCategories.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"reuseIdentifier" forIndexPath:indexPath];
+    
+    
+    CQCategory *animalObject = self.animalCategories[indexPath.row];
+    cell.textLabel.text = animalObject.name;
+    
+    cell.detailTextLabel.text = animalObject.selction;
+    
+    return cell;
+}
+
+
+#pragma mark - Navigation
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    
+    NSIndexPath *selectIndex = [self.tableView indexPathForSelectedRow];
+    
+    CQCategory *animalObject = self.animalCategories[selectIndex.row];
+    
+    DetailTableViewController *detailTable = segue.destinationViewController;
```

Comment:
Style: A better name for this variable would be `detailController` so this class doesn't make any assumptions about how the detail view is implemented (what if you wanted to use something other than a table view to implement the secondary screen?).